src/state_machine.zig

```diff
@@ -533,8 +533,41 @@ pub fn StateMachineType(
             callback(self);
         }
 
+        pub fn input_valid(operation: Operation, input: []align(16) const u8) bool {
+            switch (operation) {
+                .pulse => {
+                    if (input.len != 0) return false;
+                },
+                inline .get_account_transfers, .get_account_balances => |comptime_operation| {
+                    const event_size = @sizeOf(Event(comptime_operation));
+
+                    if (input.len != event_size) return false;
+                },
+                inline else => |comptime_operation| {
+                    const event_size = @sizeOf(Event(comptime_operation));
+                    const batch_max = comptime constants.batch_max.operation_batch_max(
+                        comptime_operation,
+                    );
+
+                    comptime assert(event_size > 0);
+                    comptime assert(batch_max > 0);
+
+                    // TODO(batiati) Clients do not validate batch size == 0,
+                    // and even the simulator can generate requests with no events.
+                    maybe(input.len == 0);
```

Comment:
Should we validate zero-sized `.create_{accounts,transfers}` and `lookup_{accounts,transfers}` ?

Reply:
I'm leaning towards it not being an error, since if it was an error that would implicitly increase the surface area of the client interface. e.g. I'm imagining someone writing client code like this:

```
var accounts_to_lookup = []
for (whatever)  {
  if (something) accounts_to_lookup.append(account);
}
client.lookup_accounts(accounts)
```

which is simpler than
```
var accounts_to_lookup = []
for (whatever)  {
  if (something) accounts_to_lookup.append(account);
}
if (accounts.len > 0) client.lookup_accounts(accounts)
```